spring-boot-project/spring-boot-actuator/src/main/java/org/springframework/boot/actuate/health/Health.java

```diff
@@ -229,6 +229,17 @@ public Builder withDetail(String key, Object value) {
 			return this;
 		}
 
+		/**
+		 * Record details using the given {@link Map}.
+		 * @param details The details as a {@link Map}
+		 * @return this {@link Builder} instance
+		 */
+		public Builder withDetails(Map<String, Object> details) {
```

Comment:
That doesn't sound right. You can pass a `Map` to the regular method and we don't want to offer a feature that would wipe out all the details (that's why you  have to use a key)